### PR TITLE
Feature/testing/use local dependencies

### PR DESCRIPTION
**Pull request by damienld22**
**Please check if the PR fulfills these requirements**
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")
```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**Scope** (check one with "x")
```
[ ] <root>
[ ] @zetapush/cli
[ ] @zetapush/client
[ ] @zetapush/cometd
[ ] @zetapush/common
[ ] @zetapush/core
[ ] @zetapush/create
[ ] @zetapush/http-server
[ ] @zetapush/platform-legacy
[x] @zetapush/testing
[ ] @zetapush/troubleshooting
[x] @zetapush/user-management
[ ] @zetapush/worker
```

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

Add possibility to use local dependencies during testing

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: packages/integration/spec/templates/default-standard-user-workflow/application.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
     }
   },
   "mailjet": {
-    "apikey-public": "f81f5b1d64a395a528e5f5c44aeba1af",
-    "apikey-private": "445647e5cf8517969b95e3363c3db7d2"
+    "apikey-public": "abb13e7529f0bb01e9768a8068017556",
```
Comment (aurelien-baudet) [inline packages/integration/spec/templates/default-standard-user-workflow/application.json]: Keys should not be committed in sources. We should provide them externally (from given for example). We should also be able to override values using environment variables (for Jenkins)

File: packages/integration/spec/testing/local-dependencies.spec.js [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+            /*   */ .and()
+            /**/ .npm()
+            /*   */ .dependencies()
+            /*     */ .modules(myLocalModules)
+            /*     */ .and()
+            /*   */ .and()
+            /**/ .apply(this);
+
+          await frontAction('call hello', this, async (api) => {
+            let failure = false;
```
Comment (aurelien-baudet) [inline packages/integration/spec/testing/local-dependencies.spec.js]: Code could be simplfied:
```javascript
try {
  const result = await api.hello();
  expect(result).toEqual('hello');
} catch(error) {
   fail('Should not have failed');
}
```
this.context.runner.stop() is not necessary here and is already done automatically by autoclean

File: packages/integration/spec/user-management/default-standard-user-workflow.spec.js [hunk 3]
```diff
@@ -107,6 +115,14 @@ describe(`As developer with
           /**/ .worker()
           /*  */ .up()
           /*  */ .and()
+          /**/ .npm()
+          /*   */ .dependencies()
+          /*     */ .module('@zetapush/user-management')
+          /*       */ .and()
+          /*     */ .module('@zetapush/core')
+          /*     */ .and()
```
Comment (aurelien-baudet) [inline packages/integration/spec/user-management/default-standard-user-workflow.spec.js]: indent

File: packages/integration/spec/user-management/default-standard-user-workflow.spec.js [hunk 4]
```diff
@@ -34,6 +34,14 @@ describe(`As developer with
...
```
Comment (aurelien-baudet) [inline packages/integration/spec/user-management/default-standard-user-workflow.spec.js]: indent

File: packages/integration/spec/user-management/default-standard-user-workflow.spec.js [hunk 5]
```diff
@@ -177,6 +193,14 @@ describe(`As developer with
...
```
Comment (aurelien-baudet) [inline packages/integration/spec/user-management/default-standard-user-workflow.spec.js]: indent

File: packages/testing/src/bdd/clean.ts [hunk 6]
```diff
@@ -2,9 +2,20 @@ import { TestContext, Test, ContextWrapper, Context } from '../utils/types';
 import { cleanLogger } from '../utils/logger';
 import { nukeApp, nukeProject } from '../utils/commands';
 import { ResolvedConfig } from '@zetapush/common';
+const execa = require('execa');
+const kill = require('tree-kill');
 
-export const autoclean = async (testOrContext: Context) => {
+export const autoclean = async (testOrContext: Context, disableResetNpmRegistry = false) => {
```
Comment (aurelien-baudet) [inline packages/testing/src/bdd/clean.ts]: `disableResetNpmRegistry` is not used. Should be removed ?

File: packages/user-management/src/standard-user-workflow/core/StandardUserWorkflowAsyncTaskService.ts [hunk 7]
```diff
@@ -48,28 +47,31 @@ export class StandardUserWorkflowAsyncTaskService extends Service {
    *  }
    * );
    * @access public
-   * @param accountDetails 
-   * @param confirmationRedirection 
+   * @param accountDetails
+   * @param confirmationRedirection
    */
-  signup(accountDetails: AccountCreationDetails, confirmationRedirection?: Redirection): Promise<PendingAccountConfirmation> {
+  signup(
```
Comment (aurelien-baudet) [inline packages/user-management/src/standard-user-workflow/core/StandardUserWorkflowAsyncTaskService.ts]: Are we going back with prettier ??
If the line hasn't changed, why formatting provides old formatting ?

File: .vscode/launch.json [hunk 8]
```diff
@@ -133,8 +133,10 @@
       // "runtimeExecutable": "~/.nvm/versions/node/v8.11.2/bin/node",
       "cwd": "${workspaceFolder}/packages/integration",
       "runtimeArgs": [
+        "-r",
```
Comment (aurelien-baudet) [inline .vscode/launch.json]: Is ts-node really needed to run integration package ?
Integration specs are all coded with js. ts-node is needed for code loaded from specs but not by integration itself ?

File: .vscode/launch.json [hunk 9]
```diff
@@ -146,7 +148,8 @@
         "ZETAPUSH_LOCAL_DEV": "true",
         "ZETAPUSH_LOG_LEVEL": "silly",
         "COMETD_LOG_LEVEL": "info",
-        "ZETAPUSH_COMMANDS_LOG_LEVEL": "-vvvv"
+        "ZETAPUSH_COMMANDS_LOG_LEVEL": "-vvvv",
+        "TS_NODE_SKIP_IGNORE": "true"
```
Comment (aurelien-baudet) [inline .vscode/launch.json]: Is this environment variable needed for launching integration ?
This environment variable seems to not be for integration itself but for projects that are run by integration. Should it be provided in tests instead (the aim is that a developer can run all tests easily, other environment variables are not mandatory to be able to run tests) ?

File: packages/integration/spec/testing/local-dependencies.spec.js (hunk 2 above)
Comment (damienld22) [reply packages/integration/spec/testing/local-dependencies.spec.js]: Done

File: packages/user-management/src/standard-user-workflow/core/StandardUserWorkflowAsyncTaskService.ts (hunk 7 above)
Comment (damienld22) [reply packages/user-management/src/standard-user-workflow/core/StandardUserWorkflowAsyncTaskService.ts]: Done

**Pull request merged**